src/fondant/pipeline/lightweight_component.py

```diff
@@ -40,6 +40,10 @@ def wrapper(cls):
             script=script,
         )
 
+        validate_required_functions_are_implemented(cls)
+        base_component_cls = cls.__bases__[0]
```

Comment:
This might not always get the correct base class.

Reply:
Good catch! I've added a separat function, iterate over all base classes and making sure that only one Fondant base class gets inherit.